### PR TITLE
feat: suspense support for usePublication hook

### PR DESCRIPTION
**Pull request by krzysu**

### CODE REVIEW

File: packages/react/src/publication/usePublication.ts [hunk 1]
```diff
@@ -26,66 +49,81 @@ export type UsePublicationArgs = OneOf<PublicationRequest>;
...
+  });
 
-  if (error) {
-    return {
-      data: undefined,
-      error,
-      loading: false,
-    };
-  }
+  // if (request.forId && !isValidPublicationId(request.forId)) {
```
Comment (krzysu) [inline packages/react/src/publication/usePublication.ts:114]: there is a problem, that with wrong publicationId the API returns error, but it's not handled anywhere so it's thrown in the example app, would it make sense to do simple publicationId validation, the same as on the API side? the other thing is, it is not possible before useSuspendableQuery hook and later is too late as the API error is already thrown. any idea?

Comment (krzysu) [reply packages/react/src/publication/usePublication.ts:114]: or that use case shouldn't really exist? I made this UI and it breaks badly when using a wrong publication Id https://lens-sdk-example-web-wagmi-q0w3lfali-avaraxyz.vercel.app/publications/usePublication

Comment (cesarenaldi) [reply packages/react/src/publication/usePublication.ts:114]: to me this is a case of wrong usage, a wrong Id (from a format perspective) to me should throw.

Comment (cesarenaldi) [reply packages/react/src/publication/usePublication.ts:114]: I rebased my branch on top of this one. I polished this aspect there. Will merge both now.

**Review by cesarenaldi — APPROVED** [review]

**Pull request merged**